Interop/Indexer/src/main/kotlin/org/jetbrains/kotlin/native/interop/indexer/J2ObjCParser.kt

```diff
@@ -0,0 +1,126 @@
+package org.jetbrains.kotlin.native.interop.indexer
+
+import org.jetbrains.org.objectweb.asm.tree.ClassNode
+import org.jetbrains.org.objectweb.asm.tree.MethodNode
+import org.jetbrains.org.objectweb.asm.Type.getArgumentTypes
+import org.jetbrains.org.objectweb.asm.Type.getReturnType
+
+// TODO: Move j2objcparser class here ;Change to org.jetbrains.org.objectweb.asm.*; Figure out gradle
+/**
+ *  Parses an ASM ClassNode and builds an ObjCClass
+ *
+ *  @param classNode An ASM classNode
+ */
+class J2ObjCParser(val classNode: ClassNode) {
+
+  /**
+   * Builds and returns an ObjCClass implementation with associated methods
+   *
+   * @return A generated ObjCClass implementation
+   */
+  fun buildClass(): ObjCClassImpl {
+    val generatedClass = ObjCClassImpl(
+      name = classNode.name,
+      isForwardDeclaration = false,
+      binaryName = null,
+      location = Location(HeaderId("")) // Leaving headerId empty for now
+    )
+    generatedClass.methods.addAll(buildClassMethods())
+    generatedClass.baseClass = ObjCClassImpl(
+      name = "NSObject",
+      binaryName = null,
+      isForwardDeclaration = false,
+      location = Location(headerId = HeaderId("usr/include/objc/NSObject.h"))
+    )
+    return generatedClass
+  }
+
+  /**
+   * Parses ASM methods and builds a list of ObjCMethods
+   *
+   * @return A list of ObjCMethods
+   */
+  fun buildClassMethods(): MutableList<ObjCMethod> {
+    val methods = mutableListOf<ObjCMethod>()
+    for (method in classNode.methods) {
+      var selector = method.name
+      if (!method.parameters.isNullOrEmpty() && method.parameters.size > 1) {
+        for (i in 1 until method.parameters.size) {
+          selector += ":" + method.parameters.get(i).name
+        }
+        selector += ":"
+      }
```

Comment:
Can probably do this in a better way

Reply:
(Builds a selector name if the method has multiple parameters)
ex: Add2Numbers(x, secondNumber) -> Add2Numbers:secondNumber